code/data_structures/linked_list/linked_lists_operations/linked_list_cycle.cpp

```diff
@@ -0,0 +1,74 @@
+/* This code detects and removes cycle in a linked list,
+	it also transforms circular lists into linear lists.
+
+	Data Entry: Beginning
+	Cycle Placement: Nth Position
+	Cycle Detection: Nth Position
+	Cycle Removal: Nth Position
+
+	By: Sarthi Chugh (user: sarthi92)
+		BITS, Pilani
+*/
+#include<iostream>
+using namespace std;
+struct Node
+{
+    int data;
+    Node* next;
+}
+*head;
+
+int linked_list_operations(int ch,int x)

```

Comment:
You should have separate functions or methods instead of cramming everything into a single function.